src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -704,6 +704,15 @@ void Unit::DealDamageMods(Unit const* victim, uint32 &damage, uint32* absorb) co
 
 uint32 Unit::DealDamage(Unit* victim, uint32 damage, CleanDamage const* cleanDamage, DamageEffectType damagetype, SpellSchoolMask damageSchoolMask, SpellInfo const* spellProto, bool durabilityLoss)
 {
+    // Sparring Checks
+    if (Creature* me = ToCreature())
+        if (Creature* target = victim->ToCreature())
+            if (me->CanSparrWith(target))
+                if (damage >= victim->GetHealth()) // safety check to prevent creatures getting accidentally killed when the percentage is small
+                    damage = victim->GetHealth() - 1;
+                else if (target->GetHealthPct() <= me->GetSparringHealthLimitPctFor(target))
```

Comment:
```
src/server/game/Entities/Unit/Unit.cpp:713:17:
 fatal error: add explicit braces to avoid dangling else [-Wdangling-else]

                else if (target->GetHealthPct() <= me->GetSparringHealthLimitPctFor(target))
                ^
```
I presume this suggests braces around the lines 711-714 in your `Unit.cpp`

Reply:
... or maybe braces are needed after the `else` (uncertain about this choice)